### PR TITLE
feat: support adding in memory files to layers

### PR DESCRIPTION
**Pull request by soldair**
BREAKING CHANGE: image.addFiles obj keys and values flipped

image.addFiles(obj)
when paths are specified as an object the keys are now `targetDirectory`.
this enables copying the same files into a container to different paths and CustomFiles

### CODE REVIEW

**Comment by ofrobots** [comment]
If you're following conventional commits style, the 'BREAKING CHANGE:' is supposed to be part of the body of the commit message.

**Comment by soldair** [comment]
ill get it in the squash ;)

File: README.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ Defined in the order that they compose into an "Image":
 ## API
 
 - `const {Image} = require('container-image-builder')`
-- or `import {Image} from 'container-image-builder'` in typescript etc.
+- or `import {Image} from 'container-image-builder'` in typescript etc.
```
Comment (DominicKramer) [inline README.md]: It looks like there is some extra whitespace at the end of the line.

File: package.json [hunk 2]
```diff
@@ -6,13 +6,14 @@
   "keywords": [],
   "scripts": {
     "test": "nyc mocha -t 30000 --require source-map-support/register build/test/*.js",
-    "test-one": "nyc mocha -t 30000 --require source-map-support/register $1",
+    "test-one": "nyc mocha -t 30000 --require source-map-support/register --grep 'packs a custom file' $1",
```
Comment (DominicKramer) [inline package.json]: I think the `--grep` should be removed so all tests are tested before this lands.

File: package.json [hunk 3]
```diff
@@ -6,13 +6,14 @@
...
     "check": "gts check",
     "clean": "gts clean",
     "compile": "tsc -p .",
     "fix": "gts fix",
     "prepare": "npm run compile",
     "pretest": "npm run compile",
+    "pretest-one": "npm run compile",
```
Comment (DominicKramer) [inline package.json]: Why is this here?

File: package.json (hunk 2 above)
Comment (soldair) [reply package.json]: great catch =)

File: package.json (hunk 3 above)
Comment (soldair) [reply package.json]: it fires before test-one
